onvif/src/soap/auth/digest.rs

```diff
@@ -31,19 +32,22 @@ enum State {
 }
 
 impl Digest {
-    pub fn new(uri: &Url, creds: &Option<Credentials>) -> Self {
+    pub fn new(uri: &Url, creds: &Option<Credentials>, reuse_headers: bool) -> Self {
         Self {
             creds: creds.clone(),
             uri: uri.clone(),
             state: State::Default,
+            reuse_headers,
         }
     }
 }
 
 impl Digest {
     /// Call this when the authentication was successful.
     pub fn set_success(&mut self) {
-        if let State::Got401 { count, .. } = &mut self.state {
+        if !self.reuse_headers {
```

Comment:
I wonder whether it's a bit easier to follow if you flip the condition and the if/else branches around.

Reply:
It's not just `else`, it is `if else`, so can't really swap the branches. But I rewrote the function, hopefully it is cleaner now.

Reply:
Sorry, I missed that this is an `if let`.